beep/protocol/maccor_to_biologic_mb.py

```diff
@@ -986,36 +994,41 @@ def _convert_steps_to_seqs(self, steps, step_num_offset, end_step_num):
         one_past_final_step_num = step_num_offset + len(steps) + 1
         goto_upperbound = one_past_final_step_num
         goto_lowerbound = step_num_offset
-        prev_loop_open = -1
+        prev_loop_open_step_num = -1
 
         seqs = []
         for i, (step, parts) in enumerate(steps_and_parts):
             # steps are 1-indexed
             step_num = step_num_offset + 1 + i
             step_type = get(step, "StepType")
             if step_type in loop_open_types:
-                prev_loop_open = step_num
+                prev_loop_open_step_num = step_num
                 # cannot goto befoer a Do
-                goto_lowerbound = prev_loop_open + 1
+                goto_lowerbound = prev_loop_open_step_num + 1
             elif step_type == "AdvCycle":
                 continue
             elif step_type in loop_close_types:
-                if prev_loop_open == -1:
+                if prev_loop_open_step_num == -1:
                     raise Exception("Unclosed loop at {}".format(step_num))
 
                 assert len(seq_nums_by_step_num[step_num]) == 1
                 seq_num = seq_nums_by_step_num[step_num][0]
+                num_loops = max(
+                    # the way maccor and biologic count loops varies by 1
+                    int(get(step, "Ends.EndEntry.Value")) - 1,
+                    0,
+                )
+                seq_num_to_loop_to = seq_nums_by_step_num[prev_loop_open_step_num][0]
```

Comment:
@patrickherring-TRI I've moved any finagling with where to loop to or how many loops here because to make the code easier to follow. This should fix the goto looping bug we discussed.